torch/distributed/rpc/functions.py

```diff
@@ -73,12 +81,54 @@ def async_execution(fn):
         >>> def async_add(to: str, x: Tensor, y: Tensor) -> Future[Tensor]:
         >>>     return rpc.rpc_async(to, script_add, (x, y))
         >>>
+        >>> # On worker0
         >>> ret = rpc.rpc_sync(
         >>>     "worker1",
         >>>     async_add,
         >>>     args=("worker2", torch.ones(2), 1)
         >>> )
         >>> print(ret)  # prints tensor([2., 2.])
+
+        When combined with static or class method, this decorator must be the
+        inner one.
+
+        >>> from torch.distributed import rpc
+        >>>
+        >>> # omitting setup and shutdown RPC
+        >>>
+        >>> # On all workers
+        >>> class AsyncExecutionClass:
+        >>>
+        >>>     @rpc.functions.async_execution
+        >>>     @staticmethod
```

Comment:
In this case the decorator `rpc.functions.async` is outer, but it looks like the doc says it needs to be inner for static methods?

Reply:
This is a vestige from debug. let me update that. Thanks for spotting this.